crates/burn-import/src/burn/node/split.rs

```diff
@@ -0,0 +1,137 @@
+use super::{Node, NodeCodegen};
+use crate::burn::{OtherType, Scope, TensorType, ToTokens, Type};
+use burn::config::Config;
+use burn::record::PrecisionSettings;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+#[derive(Config, Debug)]
+pub struct SplitConfig {
+    pub axis: usize,
+    pub num_outputs: Option<usize>,
+    pub split_sizes: Option<Vec<usize>>,
+}
+
+#[derive(Debug, Clone, new)]
+pub struct SplitNode {
+    pub input: TensorType,
+    pub outputs: Vec<TensorType>,
+    pub config: SplitConfig,
+}
+
+impl<PS: PrecisionSettings> NodeCodegen<PS> for SplitNode {
+    fn output_types(&self) -> Vec<Type> {
+        let tensor = &self.outputs[0];
+        let dims = tensor.dim;
+
+        let dims_literal = proc_macro2::Literal::usize_unsuffixed(dims);
+
+        let vec_tensor_type = quote! {Vec<Tensor<B, #dims_literal>>};
+
+        let other_type = OtherType {
+            name: syn::Ident::new("split_tensors", proc_macro2::Span::call_site()),
+            ty: vec_tensor_type,
+        };
+
+        vec![Type::Other(other_type)]
+    }
+
+    fn input_types(&self) -> Vec<Type> {
+        vec![Type::Tensor(self.input.clone())]
+    }
+
+    fn forward(&self, scope: &mut Scope, node_position: usize) -> TokenStream {
+        let input = scope.tensor_use_owned(&self.input, node_position);
+        let axis = self.config.axis.to_tokens();
+        let split_tensors = syn::Ident::new("split_tensors", proc_macro2::Span::call_site());
```

Comment:
With the above changes, we can safely unpack into a fixed number of outputs
```suggestion
        let outputs = self
            .outputs
            .iter()
            .map(|t| t.name.clone())
            .collect::<Vec<_>>();

        // We know the actual number of outputs, so we can safely unpack into a fixed number
        let unpack_outputs = quote! {
            let [#(#outputs),*] = split_tensors.try_into().unwrap();
        };
```